haystack/nodes/retriever/link_content.py

```diff
@@ -0,0 +1,202 @@
+    # TODO: implement this
+    return None
+
+
+class LinkContentFetcher(BaseComponent):
+    """
+    LinkContentFetcher fetches content from a URL and converts it into a list of Document objects.
+
+    LinkContentFetcher supports the following content types:
+    - HTML
+
+    """
+
+    outgoing_edges = 1
+
+    REQUEST_HEADERS = {
+        "accept": "*/*",
+        "User-Agent": f"haystack/LinkContentFetcher/{__version__}",
+        "Accept-Language": "en-US,en;q=0.9,it;q=0.8,es;q=0.7",
+        "referer": "https://www.google.com/",
+    }
+
+    def __init__(self, processor: Optional[PreProcessor] = None, raise_on_failure: Optional[bool] = False):
+        """
+
+        Creates a LinkContentFetcher instance.
+        :param processor: PreProcessor to apply to the extracted text
+        :param raise_on_failure: A boolean indicating whether to raise an exception when a failure occurs
+                         during content extraction. If False, the error is simply logged and the program continues.
+                         Defaults to False.
+        """
+        super().__init__()
+        self.processor = processor
+        self.raise_on_failure = raise_on_failure
+        self.handlers: Dict[str, Callable] = {"html": html_content_handler, "pdf": pdf_content_handler}
+
+    def fetch(self, url: str, timeout: Optional[int] = 3, doc_kwargs: Optional[dict] = None) -> List[Document]:
+        """
+        Fetches content from a URL and converts it into a list of Document objects.
+        :param url: URL to fetch content from.
+        :param timeout: Timeout in seconds for the request.
+        :param doc_kwargs: Optional kwargs to pass to the Document constructor.
+        :return: List of Document objects.
+        """
+        if not url or not self._is_valid_url(url):
+            raise InvalidURL("Invalid or missing URL: {}".format(url))
+
+        doc_kwargs = doc_kwargs or {}
+        extracted_doc: Dict[str, Union[str, dict]] = {
+            "meta": {"url": url, "timestamp": int(datetime.utcnow().timestamp())}
+        }
+        extracted_doc.update(doc_kwargs)
+
+        response = self._get_response(url, timeout=timeout)
+        has_content = response.status_code == HTTPStatus.OK and response.text
+        fetched_documents = []
+        if has_content:
+            handler = "html"  # will handle non-HTML content types soon, add content type resolution here
+            if handler in self.handlers:
+                extracted_content = self.handlers[handler](response, self.raise_on_failure)
+                if extracted_content:
+                    extracted_doc["content"] = extracted_content
+                    logger.debug("%s handler extracted content from %s", handler, url)
+                else:
+                    logger.warning("%s handler failed to extract content from %s", handler, url)
+                    text = extracted_doc.get("text", "")
+                    extracted_doc["content"] = text
```

Comment:
I still don't understand the meaning of this part.
Can you please explain? Why do we want to provide a default text for the document?
If really useful, it should be documented in some way.

Reply:
Valid question. See https://github.com/deepset-ai/haystack/pull/5229/files#r1261301963 for details. Whatever we agree on here as a fallback key - I'm fine

Reply:
Now I see...

What puzzles me about the current strategy is that you can end up having both `content` and `text` (inside `meta`) for the same document.

How about explicitly naming it `snippet_text` and adding a short comment explaining this?